File: src/main/webapp/assets/js/Modelo/Calle.js
/**
 * Created by Ezequiel on 03/08/2016.
 */

var Sentido = {
    NORTE_SUR: "Norte-Sur",
    SUR_NORTE: "Sur-Norte",
    OESTE_ESTE: "Oeste-Este",
    ESTE_OESTE: "Este-Oeste"
}

function Calle() { //ABSTRACTA
    this.cantCarriles = 2;
    this.sentido=undefined;
    this.cuadras = new Array();
    this.preferencia = 1;
}

function CalleVertical() {
    Calle.call(this);
    this.sentidosPosibles = [Sentido.NORTE_SUR, Sentido.SUR_NORTE];
}
createjs.extend(CalleVertical,Calle);

CalleVertical.prototype.cambiaTuSentido = function (){
    this.sentido = this.sentido == Sentido.NORTE_SUR ? Sentido.SUR_NORTE:Sentido.NORTE_SUR;
}

function CalleHorizontal() {
    Calle.call(this);
    this.sentidosPosibles = [Sentido.ESTE_OESTE, Sentido.OESTE_ESTE];
}
createjs.extend(CalleHorizontal,Calle);
CalleHorizontal.prototype.cambiaTuSentido =function (){
    this.sentido = this.sentido == Sentido.ESTE_OESTE ? Sentido.OESTE_ESTE:Sentido.ESTE_OESTE;
}

File: src/main/kotlin/com/thegrid/communication/model/Calle.kt
package com.thegrid.communication.model

/**
 * Created by Ezequiel on 20/08/2016.
 */

class Calle(cantCarriles:Int=1,
            preferencia:Int=0, sentido:String="",
            sentidosPosibles:Array<String> = arrayOf(),
            cuadras:Array<Cuadra> = arrayOf()) {
    public var cantCarriles = cantCarriles;
    public var preferencia = preferencia;
    public var sentido = sentido;
    public var sentidosPosibles = sentidosPosibles;
    public var cuadras = cuadras;
}